src/analysis/special_functions/pow.lean

```diff
@@ -601,6 +601,81 @@ end sqrt
 
 end real
 
+section measurability_real
+
+lemma real.measurable_rpow : measurable (λ p : ℝ × ℝ, p.1 ^ p.2) :=
+begin
+  have h_compl : {p : ℝ × ℝ | p.1 ≠ 0} = {p : ℝ × ℝ | p.1 = 0}ᶜ,
+  { ext,
+    rw [set.mem_compl_iff, set.mem_set_of_eq, set.nmem_set_of_eq], },
+  have h_prod : {p : ℝ × ℝ | p.1 = 0} = set.prod {(0:ℝ)} set.univ,
+  { ext,
+    rw [set.mem_set_of_eq, set.mem_prod, set.mem_singleton_iff],
+    simp, },
+  have h_meas : is_measurable {p : ℝ × ℝ | p.1 = 0},
+  { rw [h_prod, is_measurable_prod],
+    simp, },
+  refine measurable_of_measurable_union_cover {p : ℝ × ℝ | p.1 = 0} {p : ℝ × ℝ | p.1 ≠ 0} h_meas
+    h_meas.compl _ _ _,
+  { rw [h_compl, set.union_compl_self], },
+  { have h_eq_ite : (λ a : {p : ℝ × ℝ | p.fst = 0}, (a:ℝ×ℝ).fst ^ (a:ℝ×ℝ).snd) =
+      λ a : {p : ℝ × ℝ | p.fst = 0}, ite ((a:ℝ×ℝ).snd = 0) 1 0,
+    { ext1 a,
+      have h_fst_zero : (a:ℝ×ℝ).fst = 0, from a.prop,
+      rw h_fst_zero,
+      split_ifs with h_snd,
+      { rw h_snd,
+        exact real.rpow_zero _, },
+      exact real.zero_rpow h_snd, },
+    rw h_eq_ite,
+    change measurable ((λ x : ℝ, ite (x = 0) (1:ℝ) (0:ℝ))
+      ∘ (λ a : {p : ℝ × ℝ | p.fst = 0}, (a:ℝ×ℝ).snd)),
+    refine measurable.comp _ (measurable_snd.comp measurable_subtype_coe),
+    { refine measurable_of_measurable_on_compl_singleton 0 _,
+      have h_const : set.restrict (λ (x : ℝ), ite (x = 0) (1:ℝ) (0:ℝ)) {x : ℝ | x ≠ 0}
+        = λ x : {x : ℝ | x ≠ 0}, (0:ℝ),
+      { ext1 x,
+        rw set.restrict_eq,
+        change ite (x.val = 0) (1:ℝ) (0:ℝ) = (0:ℝ),
+        split_ifs,
+        { exfalso, exact x.prop h, },
+        refl, },
+      rw h_const,
+      exact measurable_const, }, },
+  { refine continuous.measurable _,
+    rw continuous_iff_continuous_at,
+    intro x,
+    change continuous_at ((λ a : ℝ × ℝ, a.fst ^ a.snd)
+      ∘ (λ a : {p : ℝ × ℝ | p.fst ≠ 0}, (a:ℝ×ℝ))) x,
+    refine continuous_at.comp _ continuous_at_subtype_coe,
+    change continuous_at (λ (p : ℝ × ℝ), p.fst ^ p.snd) x.val,
+    have h_x : x.val = (x.val.fst, x.val.snd), by simp,
+    rw h_x,
+    refine real.continuous_at_rpow_of_ne_zero x.prop _, },
```

Comment:
When you're closing a goal, `exact` is preferable to `refine`, as it shows the intent better.
```suggestion
    exact real.continuous_at_rpow_of_ne_zero x.prop _, },
```